bson/bsoncodec/map_codec.go

```diff
@@ -20,8 +22,9 @@ var defaultMapCodec = NewMapCodec()
 
 // MapCodec is the Codec used for map values.
 type MapCodec struct {
-	DecodeZerosMap   bool
-	EncodeNilAsEmpty bool
+	DecodeZerosMap         bool
+	EncodeNilAsEmpty       bool
+	EncodeKeysWithStringer bool
```

Comment:
As a thought, what if you made this a function instead and let the user provide you with something to convert when they want it different than the standard?

```
KeyEncoder func(val reflect.Value) (string, error)
```

You can provide a function that does this already, so it's just as easy as setting true.

Reply:
Due to how the packages are set up, there isn't a good way to add an interface option, so it stayed as a bool. I am open to other option names